metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/writer/ElasticsearchWriter.java

```diff
@@ -139,8 +140,12 @@ private MessageIdBasedDocument createDocument(BulkMessage<JSONObject> bulkWriter
     } else {
       LOG.warn("Missing '{}' field; timestamp will be set to system time.", TIMESTAMP.getName());
     }
-
-    return new MessageIdBasedDocument(source, guid, sensorType, timestamp, bulkWriterMessage.getId());
+    MessageIdBasedDocument messageIdBasedDocument = new MessageIdBasedDocument(source, guid, sensorType, timestamp, bulkWriterMessage.getId());
+    if (metronId) {
+      // Use the metron-generated GUID instead of letting Elasticsearch set the id
+      messageIdBasedDocument.setDocumentID(guid);
```

Comment:
Can you double-check the [debug log statements here](https://github.com/apache/metron/blob/2d2cec749c52659f4cf5b9177047b018222a13cd/metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/bulk/ElasticsearchBulkDocumentWriter.java#L91-L98) to ensure they are still correct based on this change?  I don't think they are any longer.

Reply:
Are you sure these are incorrect?  The shards are coming directly from the response so I would expect it to be accurate.

Reply:
I think I sent you the wrong link which debug logs I was referring to.  Let me see if I can't remember which ones I was thinking of.

Reply:
[These are the log statements](https://github.com/apache/metron/blob/2d2cec749c52659f4cf5b9177047b018222a13cd/metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/bulk/ElasticsearchBulkDocumentWriter.java#L129-L138) that I was thinking of.

Previously, if the document ID was not set, the document was new.  Now with these changes that may not be the case.  We probably just alter/edit/remove these comments.

Reply:
You're right, those log statements are not quite correct anymore.  I updated them in the latest commit.